api/src/routes/catalogs.rs

```diff
@@ -111,22 +114,13 @@ pub async fn get_catalogs(
 
 #[derive(serde::Deserialize, Clone, ToSchema)]
 struct CountQuery {
-    filter: Option<serde_json::Map<String, serde_json::Value>>,
-}
-impl CountQuery {
-    /// Convert filter to MongoDB Document
-    fn get_filter(&self) -> mongodb::bson::Document {
-        match &self.filter {
-            Some(f) => mongodb::bson::to_document(f).unwrap_or_default(),
-            None => mongodb::bson::Document::new(),
-        }
-    }
+    filter: Option<serde_json::Value>,
 }
 
 /// Run a count query on a catalog
 #[utoipa::path(
     post,
-    path = "/catalogs/{catalog_name}/queries/count",
+    path = "/catalogs/{catalog_name}/count",
```

Comment:
I'm not how strict we want to be with REST guidelines, but this would imply we're creating a new count for a catalog (what would that mean?) instead of creating a new count query. It's not a hard rule, but if we're going to have others depending on the raw endpoints, it might be good to name them like resources instead of actions. On the other hand, if it's just a front end or BOOM devs that will hit these endpoints, it doesn't matter as much. We could always go back to the `POST /queries/count` pattern and require catalog name in the body, making a query the resource of interest (it might be good to track queries and which user posted them). Or maybe an RPC API would be a better fit than REST?

Reply:
I think it would be good to follow Kowalski here and put these under a `queries` resource. I will merge this in, then add a few more changes on my branch.